### PR TITLE
add dns SRV based discovery

### PR DESCRIPTION
**Pull request by lburgazzoli**
I've added some basic code, let me know what should be amended, improved, etc

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/resolver/AbstractEtcdNameResolver.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import javax.annotation.concurrent.GuardedBy;
+
+import io.grpc.Attributes;
+import io.grpc.NameResolver;
+import io.grpc.ResolvedServerInfo;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+abstract class AbstractEtcdNameResolver extends NameResolver {
```
Comment (AdoHe) [inline src/main/java/com/coreos/jetcd/resolver/AbstractEtcdNameResolver.java]: I believe we should reuse this `AbstractEtcdNameResolver` with `SimpleEtcdNameResolver`.

Comment (lburgazzoli) [inline src/main/java/com/coreos/jetcd/resolver/AbstractEtcdNameResolver.java]: Once stabilized, yes definitively

File: src/main/java/com/coreos/jetcd/resolver/DnsSrvNameResolver.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    @VisibleForTesting
+    DnsSrvNameResolver(String targetUri) {
+        super(URI.create(targetUri));
+        this.name = getServiceAuthority();
+    }
+
+    @Override
+    protected List<ResolvedServerInfo> resolve() {
+        try {
```
Comment (AdoHe) [inline src/main/java/com/coreos/jetcd/resolver/DnsSrvNameResolver.java]: I think it's better to move logical here to a separate thread. Even in `SimpleEtcdNameResolver` I want to use a separate thread to update listener.

Comment (lburgazzoli) [inline src/main/java/com/coreos/jetcd/resolver/DnsSrvNameResolver.java]: Should't be better to have such logic in AbstractEtcdNameResolver ?

**Comment by lburgazzoli** [comment]
@AdoHe how does it look now ?

**Comment by AdoHe** [comment]
LGTM. @lburgazzoli thanks very much.

**Pull request merged**